### ISSUE TITLE
Is it possible to extract color of text segment?

### ISSUE THREAD

**Issue by luckymore** [ArtifexSoftware/mupdf.js#121]
just like:

```json
{
  bbox,
  text,
  font,
  // add color here
  color,
}
```

**Comment by jamie-lemon** [comment]
Not right now, I think we would need to add a color parameter to the StructuredText walk method (https://mupdfjs.readthedocs.io/en/latest/classes/StructuredText.html#walk). I think this may be possible to do, but I will refer to @ccxvii for feasibility here.

**Comment by luckymore** [comment]
awe! 
1. Actually I have completed a color Structured version based on `pdf.js`. But there are some issues with missing colors, invisible characters, textItem not in the same order as the command, not sure if you will occur the same problem about it, it's really a difficult job 🤣

2. then I have another question about text bolding that I hope you can answer.
The whole line is bolded
```js
const document = Document.openDocument(arrayBuffer, './AlphaZero-like tree-search canuide LLM decoding and training 5.pdf')
const page = document.loadPage(0)

console.log('doc', JSON.parse(page.toStructuredText().asJSON()))
```
![image](https://github.com/user-attachments/assets/ffd9346f-896f-47fe-be60-d4d26e483315)

**Comment by jamie-lemon** [comment]
Hmmmm, this is wrong, it would be better obviously if the text object had inline styling markup - we should look further into this.
What happens if you do `page.toStructuredText().asHTML()` do you get the kind of markup around the text that you might expect?

**Comment by luckymore** [comment]
Yes, `asHTML()` get the correct style, but I need the data in JSON format more than HTML

**Comment by luckymore** [comment]
I found another options to solve this:
```js
page.toStructuredText('preserve-spans').asJSON()
```

**Comment by jamie-lemon** [comment]
Sure - but I still think that we need to add something to the API to return the font color info.

**Comment by luckymore** [comment]
Yes, that'll be great! lft 👀

**Comment by eXponenta** [comment]
@jamie-lemon 
You should handle color in onChar query for text walker because inline can be different color for each char. I collecting different spans follow to font + color props.

https://github.com/ArtifexSoftware/mupdf.js/blob/master/src/mupdf.ts#L1402

**Comment by artifex-tor** [comment]
The StructuredTextWalker onChar callback receives the color of the text as an argument.
